### DISCUSSION TITLE
Monthly Challenge Ideas

### DISCUSSION THREAD

**Discussion by BekahHW** [Virtual-Coffee/VC-Community-Docs#144] [Monthly Challenges]
@Virtual-Coffee/monthly-challenge 
Let's start some brainstorming on what kind of challenges we can come up with.

A couple on the list that we haven't done:

- Writing a talk
- Documentation
- Accessibility

**Comment by jonathanyeong** [comment]
Love this idea! Do we have a list of monthly challenges that we've done?

**Reply by ClJarvis** [reply]
Oct - HacktoberFest
Nov - 50,000 words group blog challenge (NaNoWriMo)
Dec -
Jan - 
Feb -
March -
April -
May - Feedback

There was resume/portfolio improvement, co-working,

**Reply by vanessacor** [reply]
Oct - HacktoberFest
Nov - 50,000 words group blog challenge (NaNoWriMo)
Dec - Month of Pairing
Jan - Month of Learning
Feb - Month of Sharing
March - Get the job ready
April - Month of Kindness
May - Feedback

**Reply by danieltott** [reply]
Here's the full list for reference! 

https://virtualcoffee.io/monthlychallenges/

**Reply by jonathanyeong** [reply]
Thank you!

**Comment by vanessacor** [comment]
I love these ideas!
Here some more to add to this brainstorming
- Collaboration - work together, could be something like pair programming or get together to solve an issue or add a feature to one of the open-source VC repos for example. 
- Self-care / health - here could be start a new healthy habit, set a goal for the month

**Reply by BekahHW** [reply]
@vanessacor I updated your list above with Month of Pairing (December). That actually worked pretty well. It was nice to see a lot of collaboration.

Karen D. has a spreadsheet for building a piano playing habit--or at least that's what I'm using it for. lol. I could see tracking like that.

**Reply by vanessacor** [reply]
That's great, we can share it so people can use it.  Another idea, for people who want some accountability, we could adapt and create a group tracking habits spreadsheet. Everyone who wants to participate on the group one adds their name and habit to track, this way we could check on everyone progress.

**Comment by bogdaaamn** [comment]
I love @vanessacor's **self-care idea** of starting a new healthy/self-care habit. Also providing support and accountability would be wonderful. 

I would see the benefit of an **accessibility challenge**. Something like doing a11y changes to a codebase, or reading about it, or raising awareness. 

I love your ideas!

**Reply by BekahHW** [reply]
We have a couple of members who have agreed to help with that. I just want to make sure we have a really solid plan for accessibility that's open to everyone.

**Comment by jonathanyeong** [comment]
Love @vanessacor's ideas! Self-care is so important!!! A couple more to think about:

- Experimentation - picking up and trying new languages/frameworks. There's a Google sheet circulating somewhere with a list of VC skillz. Encouraging pairing with other people that work in a different language.
- Build in public - Start the month with a project you want to work on and post updates throughout the month on your progress.

**Reply by bogdaaamn** [reply]
**Build in public** yessssss I love this!!

**Reply by BekahHW** [reply]
I really like build-in public, @jonathanyeong! I think it adds to the things we've done like pairing, sharing, and feedback. And it would open things up.

We could build a challenge around participating in a type of daily/weekly standup. This could be async/snyc/or both.

I've been trying to figure out how we can encourage collaborative projects that could potentially move into hacktoberfest oss projects if ppl/teams wanted that. 

This could be a space for lots of different ppl to work on projects that build skills, resumes, collab, and for others to talk through the things they're working on.

Some people mentioned that maybe starting something like a daily standup would help with that. People could naturally see what others are working on and work out a collaboration.

I'd be interested in how we would break this into monthly theme and challenge separately. Anyone have thoughts on that?

**Reply by jonathanyeong** [reply]
Yes!!! Maybe the challenge could be a lightning talk on all the things you've learned/built throughout the month. With a focus on the learnings because let's face it a month isn't a long time to build a project from start to finish (if that's the route people take). EDIT: Maybe the talk could be like "here's what I knew at the start. Here's what I know now"

Also, thinking about that last line. If people didn't have a project that they wanted to build, maybe we could encourage them to contribute to an open source project run by a VCer (e.g. virtual coffee, protege, forem). And have office hours to help these people make those contributions.

Final thought! We could also encourage groups of people to build on a single project. Maybe initially we could have people share ideas and if they want help on a particular idea. I do have some concerns, I don't want this group project to turn out like group work at uni or bootcamp. That wouldn't be a fun experience. But it might encourage people who are just starting out and have no idea what/how to build something to join into the challenge.

**Reply by vanessacor** [reply]
Love this build-in public idea and it could grow into other themes, like accessibility and documentation. 
People could choose to:
- contribute to VC open source projects
- start a new project with other members
- document their own project

**Comment by tkshill** [comment]
Warning: messy thought dump approaching

I love the build in public idea as well. Current thoughts in my head.

#### What does success look like?
- I feel like it's somewhere along the lines of "people who would not have pursued a project with public accountability actually do so, and make measurable progess."
- What can we do as a community that will take people from "I'm too nervous/scared/overwhelmed/confused/apathetic" to start this build in public, to now I can? 
- Before we've done weekly milestones, but I'm wondering if we can keep the tangible steps, but decouple them from the weeks, so people don't feel behind if they start late? Like, there's four goals and you have the month and we all support each other through those steps of the journey.
- A platform for accountability feels important. VC can be that space, where all participants Say what they're working on what their plans are. We used to make spaces on the site for the early monthly themes. That might be cool to do again.
- I love the idea of forming groups and pairs @jonathanyeong. Lots of folks here have been forming ad hoc accountability pairings and it seems to be working well. Even if people don't work on the same thing, just having someone who checks in on your progress is nice.
- Similar to hacktoberfest, having some go to things for folks to get into might be helpful. We can re-up our open source projects, and highlight other practical suggestions
- The phases I'm thinking of
    - Part 1: pick your project. pick a SMART goal and a reasonable number of public checkins. Find an accountabilty partner/group. Set an MVP and a target to hit. (We can help folks decide those)
    - Part 2: Just do that first bit of work. Starting is always hardest and accountability partners can choose days and times where they start together, to make it more likely to follow through. Post when you progress. Even if it's just a message in the slack. Or we can do something similar to the tech-study-interview-group and let people present on a weekly basis.
    - Part 3: Finish your MVP
    - Part 4: Present! blog, video, stream, slides, some way to wrap up what you did

####  General challenge/theme structure
There's a greater convo here about the structure of monthly challenges. There are definitely some things we want to recur, and maybe there's a healthy cycle of Build -> Show -> Recover that this ties into? We have a month or two dedicated to Doing a thing, and then a month dedicated to Sharing what you did (with lightning talks as the finisher), and then a month of Appreciating.  Having the cycle also ensures that new members are never too far away from participating in the experiece.

And we can sort potential monthly themes into "doing", "sharing", and "caring" pools. 

Can you tell I have had way too much coffee today?

**Reply by BekahHW** [reply]
I love so much of this, but my concern with doing this cycle is that we're constantly getting new members. How do we onboard them into this cycle?

**Reply by tkshill** [reply]
I think there's a way to have the large theme but still allow people to jump in.

If you join during a writing month, you can still write on something you've been working on or wanted to get into. "Caring" months also don't require any setup. The months can stand alone but we just pick the themes in a cycle to avoid burnout

I think the hardest month would be the month of hacktoberfest since you wouldve missed the prep from the month before that, but we always have the docs and videos available?

To avoid overreach, I feel like it would be 3 cycles max

So:
- January - Learning
- Febraury - Building 
- March - Sharing (lightning talk)
- April - Caring, Health, Empathy (VC birthday month)
- May - Learning
- June - Building
- July - Sharing (lightning talk)
- August - Caring, Health, Empathy
- September - Hacktoberfest
- October - Hacktoberfest
- Novermber - Sharing (lightning talk)
- December - Caring, Health, Empathy (happy hour month)

We may not even need to highlight the cycle too much to new members, except to let them know that if they missed a challenge that seems interesting earlier in the year, it'll definitely come up gain

**Reply by BekahHW** [reply]
This looks really interesting, and I like it. I want to think about it some more.

Doing Hacktoberfest back to back with lightning talks almost killed me last year. So I want to make sure that we recognize that support structures need to be in place for anyone who is acting as leads and support on both of these things.

**Reply by vanessacor** [reply]
This is a interesting idea, and this case we could try to form different teams for each block (one for learning, another for building...) and Hacktoberfest all hands.

**Comment by ClJarvis** [comment]
For Stand up: At my bootcamp we used a slack bot on the non-class nights. It set to a certain time, ask the stand up questions then we would reply. It was mainly to keep us accountable to our selves.

**Comment by abbeyperini** [comment]
Challenge theme proposal: Art/Creative
- Like healthy habits, promoting hobbies/self-expression not strictly related to our jobs.
- We've got several musicians, and I've seen Ryan Lynch and Mark Noonan post recordings of music.
- Chris Jarvis has his CSS art and when I floated the idea in #making-stuff, Rafi mentioned ASCII art.
- Drew Clements and his graphics. 
- Quite a few fiber artists.
- Could have a virtual show at the end of the month.

**Reply by BekahHW** [reply]
"Creative Community" Challenge?

**Reply by bogdaaamn** [reply]
> Could have a virtual show at the end of the month.

Can we have a karaoke party? 😬

**Reply by ClJarvis** [reply]
"Creative Community" Challenge? is a good descriptive title. Tells what we will be doing.

Karaoke New Years party.
